src/main/drivers/max7456.c

```diff
@@ -479,15 +515,18 @@ void max7456DrawScreenPartial(void)
 
         //------------   end of (re)init-------------------------------------
 
-        for (k=0; k< MAX_CHARS2UPDATE; k++) {
-            if (screenBuffer[pos] != shadowBuffer[pos]) {
+        // Note that k should not be incremented on each
+        // iteration, only when we find a character that
+        // we need to send to the OSD.
+        for (k=0; k< MAX_CHARS2UPDATE;) {
```

Comment:
This needs to be profiled to ensure `max7456DrawScreenPartial` doesn't block for too long

Reply:
I'll leave that up to you, since I don't have an STLink available ATM.

Reply:
Will do

Reply:
Do we have a macro for alignment? I can't find it, but I might be looking in the wrong places (or  grepping for the wrong strings, should I say).

If we don't, I'd like to introduce one in `common/utils.h`, since writing bare attributes looks a bit  dirty to me.

Reply:
No, we don't have a macro for alignment